@uportal/context-menu-navigation/src/components/ContextMenu.vue

```diff
@@ -169,14 +169,59 @@ export default {
             );
         },
         portletContent() {
-            return this.registry
-                ? this.registry.find(portlet => portlet.fname === this.selected)
-                : {};
+            const selDetails = this.contentItemDetails;
+            if (selDetails) {
+                const retVal =
+                    selDetails.parameters && selDetails.parameters.widgetType ? selDetails : false;
+                this.logDebug(
+                    'portletContent: [' +
+                        this.selected +
+                        '].  retVal=[' +
+                        retVal +
+                        '].  Selected details:'
+                );
+                this.logDebug(selDetails);
+                return retVal;
+            } else {
+                if (this.debug)
+                    console.log(
+                        'portletContent: [' + this.selected + '].  No selected details found.'
+                    );
+                return false;
+            }
         },
         contentRegistry() {
             return this.tab.content
                 .reduce((collection, group) => [...collection, ...group.content], [])
                 .map(c => c.fname);
+        },
+        contentItemDetails() {
+            if (this.selected) {
+                return this.tab.content
+                    .reduce((collection, group) => [...collection, ...group.content], [])
+                    .find(c => c.fname == this.selected);
+            } else {
+                return this.selected;
+            }
+        },
+        cleanWidgetTemplate() {
+            const escapedXmlMapping = {
+                '&lt;': '<',
+                '&gt;': '>',
+                '&quot;': '"',
+                '&amp;': '&'
+            };
+            // Assumes portletContent is truthy, thus cid will be defined
+            const cid = this.contentItemDetails;
+            if (cid && cid.parameters && cid.parameters.widgetTemplate) {
```

Comment:
You may want to try this out, not sure if babel is updated enough to support it yet, but it is easier to read if it does
```suggestion
            if (this.contentItemDetails?.parameters?.widgetTemplate) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining